### ISSUE TITLE
LGC2 screen randomly turns on.

### ISSUE THREAD

**Issue by P00NtANG** [JPersson77/LGTVCompanion#80]
I recently updated to v1.8 from v1.7.

But during the middle of the night I noticed the screen came out of stand-by on it's own and was displaying the built-in screen saver images that auto activates when there is no detected input. 

This happened two nights in a row, so I uninstalled v1.8 and went back to v1.7 and it has not happened since.

A bug maybe?

**Comment by P00NtANG** [comment]
Hi, no I just manually switch between inputs as required.
But I can certainly install v1.8 again and enable logging, no problem. 👍

**Comment by ryanmoses** [comment]
> I recently updated to v1.8 from v1.7.
> 
> But during the middle of the night I noticed the screen came out of stand-by on it's own and was displaying the built-in screen saver images that auto activates when there is no detected input.
> 
> This happened two nights in a row, so I uninstalled v1.8 and went back to v1.7 and it has not happened since.
> 
> A bug maybe?

Hi there.  Thank you @JPersson77 for this great software.   Unfortunately I am seeing this issue also.  There have been no changes that I'm aware of outside of some LG automatic updates in the last few days.  

I am a new user with v1.8 so I'm unable to compare behaviors with 1.7 however my LG C2 is constantly turning itself on in the last few days.    EG  I turn off the TV & shut my PC down and the TV will turn back on.  

I've enabled logging in case it can help.

Thanks!

**Comment by ryanmoses** [comment]
Hi - I was having a similar issue but after enabling logging I found my TV had changed IP addresses.  Unsure if this helps. 

Thanks!

**Comment by JPersson77** [comment]
@ryanmoses thanks too!

@P00NtANG any feedback wrt this issue?

**Comment by hsuanguo** [comment]
I have seen this twice in the last 2 days, found the TV on with no signal input and playing the built-in images

**Comment by JPersson77** [comment]
Hi @hsuanguo can you please enable logging and see if you can capture the issue so I can have a look. Not sure from your or @P00NtANG s description if the PC was on at the time etc.

**Comment by hsuanguo** [comment]
> Hi @hsuanguo can you please enable logging and see if you can capture the issue so I can have a look. Not sure from your or @P00NtANG s description if the PC was on at the time etc.

Sure, I'll enable it and if it captures the failure again I'll update it here.

**Comment by ryanmoses** [comment]
> @ryanmoses thanks too!
> 
> @P00NtANG any feedback wrt this issue?

Hi @JPersson77 

Sorry for the delay - I have been meaning to get back to you.

I still have some odd behavior where the TV will power off or on unexpectedly but I feel like it's a problem with windows 11 monitor topology rather than LG TV companion.

EG I have three monitors and last night I switched off two of them while playing a game and kept my OLED on. OLED was the primary display. However turning off the other monitors caused my OLED to shut down also.  Windows was pretty confused about the monitor situation when I tried to turn it back on  (main display switched several times and swapped to the wrong monitor).

I've got logging switched on and I think this is the log of the issue I describe above.  Note I did not change my monitor topology at all during this time.

The last entry in the log is PC shutdown.

```
Tue 21:12:57 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:12:57 > IPC, windows monitor topology was changed.
Tue 21:12:57 > Device1, spawning DisplayPowerOnThread().
Tue 21:12:57 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:12:57 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:12:57 > Device1, established contact: Display is ON.  
Tue 21:12:58 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:12:58 > IPC, windows monitor topology was changed.
Tue 21:12:58 > Device1, spawning DisplayPowerOnThread().
Tue 21:12:58 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:12:58 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:12:58 > Device1, established contact: Display is ON.  
Tue 21:12:59 > Device1, repeating WOL broadcast ended
Tue 21:13:00 > Device1, repeating WOL broadcast ended
Tue 21:34:28 > IPC, windows monitor topology was changed.
Tue 21:34:28 > Device1, spawning DisplayPowerOffThread().
Tue 21:34:28 > Device1, established contact: Device is OFF.
Tue 21:34:29 > IPC, windows monitor topology was changed.
Tue 21:34:29 > Device1, spawning DisplayPowerOffThread().
Tue 21:34:29 > Device1, established contact: Device is OFF.
Tue 21:34:52 > IPC, windows monitor topology was changed.
Tue 21:34:52 > Device1, spawning DisplayPowerOffThread().
Tue 21:34:52 > Device1, established contact: Device is OFF.
Tue 21:35:09 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:35:09 > IPC, windows monitor topology was changed.
Tue 21:35:09 > Device1, spawning DisplayPowerOnThread().
Tue 21:35:09 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:35:09 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:35:10 > Device1, established contact: Display is ON.  
Tue 21:35:11 > Device1, repeating WOL broadcast ended
Tue 21:35:17 > IPC, windows monitor topology was changed.
Tue 21:35:17 > Device1, spawning DisplayPowerOffThread().
Tue 21:35:17 > Device1, established contact: Device is OFF.
Tue 21:35:18 > IPC, windows monitor topology was changed.
Tue 21:35:18 > Device1, spawning DisplayPowerOffThread().
Tue 21:35:18 > Device1, established contact: Device is OFF.
Tue 21:35:20 > IPC, windows monitor topology was changed.
Tue 21:35:20 > Device1, spawning DisplayPowerOffThread().
Tue 21:35:20 > Device1, established contact: Device is OFF.
Tue 21:35:21 > IPC, windows monitor topology was changed.
Tue 21:35:21 > Device1, spawning DisplayPowerOffThread().
Tue 21:35:21 > Device1, established contact: Device is OFF.
Tue 21:35:23 > IPC, windows monitor topology was changed.
Tue 21:35:23 > Device1, spawning DisplayPowerOffThread().
Tue 21:35:23 > Device1, established contact: Device is OFF.
Tue 21:35:31 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:35:31 > IPC, windows monitor topology was changed.
Tue 21:35:31 > Device1, spawning DisplayPowerOnThread().
Tue 21:35:31 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:35:31 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:35:31 > Device1, established contact: Display is ON.  
Tue 21:35:33 > Device1, repeating WOL broadcast ended
Tue 21:35:42 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:35:42 > IPC, windows monitor topology was changed.
Tue 21:35:42 > Device1, spawning DisplayPowerOnThread().
Tue 21:35:42 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:35:42 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:35:42 > Device1, established contact: Display is ON.  
Tue 21:35:44 > Device1, repeating WOL broadcast ended
Tue 21:35:51 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:35:51 > IPC, windows monitor topology was changed.
Tue 21:35:51 > Device1, spawning DisplayPowerOnThread().
Tue 21:35:51 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:35:51 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:35:51 > Device1, established contact: Display is ON.  
Tue 21:35:53 > Device1, repeating WOL broadcast ended
Tue 21:36:00 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:36:00 > IPC, windows monitor topology was changed.
Tue 21:36:00 > Device1, spawning DisplayPowerOnThread().
Tue 21:36:00 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:36:00 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:36:00 > Device1, established contact: Display is ON.  
Tue 21:36:02 > Device1, repeating WOL broadcast ended
Tue 21:36:09 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:36:09 > IPC, windows monitor topology was changed.
Tue 21:36:09 > Device1, spawning DisplayPowerOnThread().
Tue 21:36:09 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:36:09 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:36:09 > Device1, established contact: Display is ON.  
Tue 21:36:11 > Device1, repeating WOL broadcast ended
Tue 21:36:18 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:36:18 > IPC, windows monitor topology was changed.
Tue 21:36:18 > Device1, spawning DisplayPowerOnThread().
Tue 21:36:18 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:36:18 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:36:18 > Device1, established contact: Display is ON.  
Tue 21:36:19 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:36:19 > IPC, windows monitor topology was changed.
Tue 21:36:19 > Device1, spawning DisplayPowerOnThread().
Tue 21:36:19 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 21:36:19 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 21:36:19 > Device1, established contact: Display is ON.  
Tue 21:36:20 > Device1, repeating WOL broadcast ended
Tue 21:36:21 > Device1, repeating WOL broadcast ended
Tue 22:24:47 > IPC, Device1 - enabled in windows monitor topology.
Tue 22:24:47 > IPC, windows monitor topology was changed.
Tue 22:24:47 > Device1, spawning DisplayPowerOnThread().
Tue 22:24:47 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 22:24:47 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 22:24:47 > Device1, established contact: Display is ON.  
Tue 22:24:49 > Device1, repeating WOL broadcast ended
Tue 22:24:52 > IPC, Device1 - enabled in windows monitor topology.
Tue 22:24:52 > IPC, windows monitor topology was changed.
Tue 22:24:52 > Device1, spawning DisplayPowerOnThread().
Tue 22:24:52 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 22:24:52 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 22:24:52 > Device1, established contact: Display is ON.  
Tue 22:24:54 > Device1, repeating WOL broadcast ended
Tue 22:26:50 > IPC, Device1 - enabled in windows monitor topology.
Tue 22:26:50 > IPC, windows monitor topology was changed.
Tue 22:26:50 > Device1, spawning DisplayPowerOnThread().
Tue 22:26:50 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 22:26:50 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 22:26:50 > Device1, established contact: Display is ON.  
Tue 22:26:51 > IPC, Device1 - enabled in windows monitor topology.
Tue 22:26:51 > IPC, windows monitor topology was changed.
Tue 22:26:51 > Device1, spawning DisplayPowerOnThread().
Tue 22:26:51 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 22:26:51 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 22:26:51 > Device1, established contact: Display is ON.  
Tue 22:26:52 > Device1, repeating WOL broadcast ended
Tue 22:26:53 > Device1, repeating WOL broadcast ended
Tue 23:32:07 > IPC, Device1 - enabled in windows monitor topology.
Tue 23:32:07 > IPC, windows monitor topology was changed.
Tue 23:32:07 > Device1, spawning DisplayPowerOnThread().
Tue 23:32:07 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 23:32:07 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 23:32:07 > Device1, established contact: Display is ON.  
Tue 23:32:09 > Device1, repeating WOL broadcast ended
Tue 23:32:12 > IPC, Device1 - enabled in windows monitor topology.
Tue 23:32:12 > IPC, windows monitor topology was changed.
Tue 23:32:12 > Device1, spawning DisplayPowerOnThread().
Tue 23:32:12 > Device1, repeating WOL broadcast started to MAC: 80:5B:65:7D:E3:FA using IP address: 10.0.0.38
Tue 23:32:12 > GetBestRoute2() selected interface index 13 LUID 1689399733518336 route protocol 2
Tue 23:32:12 > Device1, established contact: Display is ON.  
Tue 23:32:14 > Device1, repeating WOL broadcast ended
Tue 23:52:10 > Event subscription callback: system shut down detected.
Tue 23:52:18 > ** System is shutting down.
Tue 23:52:18 > Device1, spawning DisplayPowerOffThread().
Tue 23:52:18 > Device1, established contact: Device is OFF.
Tue 23:52:18 > The service terminated.
```

**Comment by JPersson77** [comment]
Hi @ryanmoses, I'm not sure this is the same issue as @P00NtANG  reports (need more info there) but let's assume it might be. Since I don't have the beginning of your log I will assume there is only one WebOS-device/OLED, Device1.

Your log looks quite normal in the sense that the app does what it is supposed to. So a couple of questions:

- I notice that at some points the topology seem to change in very rapid succession, two events within a second. Not sure why but maybe you recognise this and can explain what happened there? The app and display seem to be processing it fine though
- can you try and verify that the actual topology as set in windows is what is actually being logged? I e whenever the OLED is set to enabled in the windows options you should see two lines like this in the log:

```
Tue 21:12:57 > IPC, Device1 - enabled in windows monitor topology.
Tue 21:12:57 > IPC, windows monitor topology was changed.
```

and then when the OLED is set as inactive in the topology there should be only this message in the log: 
`Tue 21:34:28 > IPC, windows monitor topology was changed.`

In other words, what we must ultimately try to determine is if any of the above events in your log noted as 'IPC, windows monitor topology was changed' are erroneous. So that even though there was not an actual change in the windows topology the app still received a windows message that the topology was changed. I mean that your error could potentially pop up, if the app is registering events which are not truly user initiated. So please help me more to get a grip on the issue :)

**Comment by JPersson77** [comment]
Oh, and also. Are you using a 2022 model OLED, C2, G2 etc or an earlier model? Do you have "Always Ready" enabled if so?

**Comment by ryanmoses** [comment]
> Oh, and also. Are you using a 2022 model OLED, (C2, G2 etc) or an earlier model? Do you have "Always Ready" enabled if so?

Hi @JPersson77 - Thanks for your reply.  

I am using a 2022 LG C2 and "always ready" is currently disabled.  I have tried with this both enabled/disabled.  And yes there is only `Device1`.

I spotted those frequent topoIogy changes in the log as well but note I didn't change the topology at all during that time. 

And as mentioned I'm getting the feeling my windows 11 installation is pretty confused about the topology and it may not be anything to do with LG TV companion.     It's an upgrade of an old Windows 10 installation and likely has a few cobwebs.  

I think if you don't spot any smoking guns in the log it may be best to leave this with me.   I will reinstall windows at some point and connect my TV via cable rather than WIfi - Presumably things will settle down after that and if not I can get in touch with you again.

**Comment by JPersson77** [comment]
Hi, I'm surprised it works at all without "Always Ready" enabled. I've noted it as requirement for using this feature as my testing indicated that the nVidia control panel could not properly enable the OLED display otherwise. 

Anyway, yes I think there might be an issue here since you have logged received windows messages indicating topology changes even though you did not actually change the topology. I'll think about it a bit more :)

**Comment by JPersson77** [comment]
> > Hi, I'm surprised it works at all without "Always Ready" enabled. I've noted it as requirement for using this feature as my testing indicated that the nVidia control panel could not properly enable the OLED display otherwise.
> > Anyway, yes I think there might be an issue here since the app have logged received windows messages indicating topology changes even though you did not actually change the topology. I'll think about it a bit more :)
> 
> I am using C2 2022 and can confirm my "Always Ready" is disabled as well, and LGTV Companion works just fine most of the time, just experienced 2 times that TV was not off so far...

That's interesting. are you using an AMD or Intel GPU possibly? Can you also confirm you have the "windows topology" feature enabled, well just for clarity?

**Comment by JPersson77** [comment]
PS I created a discord server if you care to have a more direct discussion on the potential issue  https://discord.gg/7KkTPrP3fq

**Comment by JPersson77** [comment]
There is a development build v1.8.5 available on discord. This is a bugfix build with some changes to the power on/off logic that may resolve some issues noted in this thread. If you download and install this build please provide feedback on any observed changes in ability to keep power state between PC and display in sync and the specific fixes noted below. 

- bugfix for reported issues with random power on/off potentially related to wm_displaychange message handling in the v1.8 topology feature
- made clear that change of GPU and / or adding displays may invalidate the topology configuration. 
- Implemented detection for invalidated topology configuration which will show windows notification and log message urging user to reconfigure.
- added mechanism for identifying invalid pairing keys
-  more help texts was added to all configuration options in the UI

https://discord.gg/7KkTPrP3fq

**Comment by hsuanguo** [comment]
I just saw the built-in screen saver again (13:59)

```
Sat 10:18:24 > ** System requests displays OFF.
Sat 10:18:24 > Device1, spawning DisplayPowerOffThread().
Sat 10:18:24 > Device1, HDMI4 input active.
Sat 10:18:24 > Device1, established contact: Device is OFF.
Sat 10:43:58 > ** System requests displays ON.
Sat 10:43:58 > Device1, spawning DisplayPowerOnThread().
Sat 10:43:58 > Device1, repeating WOL broadcast started to MAC: ******************** using IP address: *********
Sat 10:43:58 > GetBestRoute2() selected interface index 28 LUID ********************** route protocol 2
Sat 10:44:00 > GetBestRoute2() selected interface index 28 LUID ********************** route protocol 2
Sat 10:44:01 > Device1, established contact: Display is ON.  
Sat 10:44:02 > Device1, repeating WOL broadcast ended

```

But there is no logging information at all, the last one was 3 hours ago().

By the way, I am using nvidia GPU, what is `windows topology` feature? I don't see in the UI?

**Comment by JPersson77** [comment]
Hi @hsuanguo

If there is nothing in the log for what happened at 13.59 I doubt the issue you are seeing have to do with the app.

But can you clarify please so I understand. From your earlier note here I assume the PC was on at 13.59 and I also read that the PC was at this time idle and displays were off.  Then all of a sudden the display wakes up and since there is no output from the PC the built in screensaver is playing. Can you confirm this is the issue you are experiencing? 

But if the excerpt of the log you pasted above is the last part of the log (I think that is what you write) I don't see that the app has turned off the screen when the PC became idle. Did you shut off the TV with the remote or how did it come to turn off? 

Consider if there is anything else waking the screen? Consoles or other devices using CEC? Another installation of lgtv companion on another computer which is configured to control this device? Cat stepping on remote? :)

The topology feature is new in version 1.8.0. It's not specific to any brand of GPU. The option can be found in the general options at the bottom. Previous versions only synced power state of all displays with the power state of the PC. This new feaature additionally syncs enabled/disabled state in the monitor configuration with the display power state. I e so that if you have multiple monitors connected to your PC and then disable an OLED in your monitor configuration the corresponding OLED device will now also power off. So that the OLED behave like a PC-monitor in this respect too.

**Comment by hsuanguo** [comment]
Hi @JPersson77 
Yes, the PC was idle and there was no display output from the PC, I am not sure if the screen ever went off then woke up, what happened was, I left my pc for a while, when I came back to it, I saw the TV is on with built-in screen saver.

I have no other device that controls the TV, neither did I(nor anything, anyone) use the remote.

Then I moved the mouse, then the pc woke up, worked as usual.

**Comment by hsuanguo** [comment]
[Log.txt](https://github.com/JPersson77/LGTVCompanion/files/10200834/Log.txt)
[config.json.txt](https://github.com/JPersson77/LGTVCompanion/files/10200835/config.json.txt)

Thank you very much, really appreciate on your quick response.
Please find the uploaded file above.

It just happened again(20:00) I guess, the TV was off just now, but I can't wake it up from PC, instead I have to use the remote.
I suppose it went to screen saver first, then turned itself off after a period. Again, there is no log in the log file.

I do see LGTVsvc.exe is running, and I can control manually with test function.


PC is desktop, GPU is RTX 3090. Windows 11 21H2, pc is configured to turn screen off after 10 minutes, no sleep at all.

Hope those information helps.

**Comment by hsuanguo** [comment]
I have installed 1.8.5 I'll let you guys know if it makes differences

**Comment by JPersson77** [comment]
Thanks, looking at the log I see nothing that  immediately stands out. 

But I do notice that at the times you noted the issue, 13.59 and 20.00, the log shows several hours of system being in the ON-state. I assume this means that you were mostly using the PC and the 10 minutes screen saver never kicked in. 

So here's another thought. Can you have a look at what value you have set for the display auto power off, i e this setting: et Settings->General->OLED Care->Device Self-Care->Energy Saving->Auto Power off. I think there is a potential for seeing your issue if you have this option set to 2 hours for example. 

So, hypothetically, what could happen is this: say you've been using the PC for close to two hours or something and just as you step out for a quick break the TV is automatically powering off itself. The app does not know this as windows does not know this. So this can lead to you coming back to screen that is powered off which you cannot wake with the mouse because windows beleives the screen is already on and does not send the "wake up"- message that the app operates on. There would also be nothing in the log, like what is observed. And here is an interesting thing - depending on other settings there is a fringe case in v1.8 and earlier where the app tell the screen to power off but it will instead cause it to wake up. So, again hypothetically, if your display powered off automatically on its own accord due to the above setting, then followed shortly thereafter by windows going into idle and shutting down display outputs the app would tell the display to shut down, but instead accidentally waking it up and thus showing the screen saver images you described.

If you have the displays' own auto-power off set to two hours consider increasing it to 8 hours or something and maybe consider using the idle detection of the app instead set to 5 minutes or something like that.

The issue about accidentally waking up the screen when it should be powered off was fixed in v1.8.5 by querying the TV for the status before issuing the command. Thus adding a small roundtrip delay depending on the LAN to the power off which I hope should not matter.

Let me know how it goes

**Comment by hsuanguo** [comment]
It just happened again with 1.8.5, so must be something with my system configuration. Do you know if there is a way that I can check the system log if there was a screen off event?
I do have the TV auto power off set to 4 hours, but I am pretty sure this is not the case because this time I just used the remote 20 mins ago(and switched back to PC screen) before I saw the screen saver again.......

**Comment by JPersson77** [comment]
Hi, ok I thought we had nailed it :)

One thing though, I suspect the auto power off will power off your TV after 4 hours irrespective of whether you have used the remote or not. So maybe not discount it yet. However with the fix in v1.8.5 I would expect the display to be off but not possible to wake with the mouse if the theory was correct.

Interesting question about the system log (event log). This tool will show you when PC started/went to sleep: https://www.nirsoft.net/utils/computer_turned_on_times.html . I don't think however there is a straightforward way of reading display on/off event in the the event log. I just now used this tool https://www.nirsoft.net/utils/full_event_log_view.html which presents event log items flattened so it's easy to see/filter/search what events occured at a specific time. Maybe you can find something? The closest I can see  from a brief look is that message 1660 and 1661 related to windows audio seem to occur in parallell with display on/off (on this system)

**Comment by JPersson77** [comment]
Hi @hsuanguo any feedback wrt the above? Still the same?

**Comment by JPersson77** [comment]
OK thanks for feedback and pointing out the screensaver. I will try to see if I can replicate

**Comment by hsuanguo** [comment]
> OK thanks for feedback and pointing out the screensaver. I will try to see if I can replicate

OK, I think now I have a better understanding on what happned:

- Windows settings:  screen saver on (1min), screen off (10 mins)

- Procedure:
  -  First, windows will trigger screen saver after 1min afk.
  - Then, a few minutes later(< 10min), GPU will stop ouput signal, from the TV, it will detect no signal input, so TV built-in screen saver will start. I don't really no if this is expected windows behaviour, but I am not aware of any settings regarding this.
  - If I wait a bit longer the TV will be turned off and I can see `> ** System requests displays OFF.` in logger.

So I think from LGTVCompanion side, everything works just fine as it should be.

Sorry for all the confusions,  and thanks for all the help and sugguestions you provided!

**Comment by JPersson77** [comment]
Thanks for helping out too! 

I think that is an unexpected behavior you have observed! I would expect GPU to cut signal as a result of OS requesting it to, so a bit surprising that it would do that <10mins. Not sure why and I cannot recreate at the moment on my system. nor does google give me any more info. I think build 1.8.5 and later will help somewhat with the issue since it will avoid potentially powering on the screen inadvertently. 

If you should come to think of anything more related to this please let me know

**Comment by JPersson77** [comment]
Closing this issue since there was no potential fix to the app identified (apart from what was implemented in v1.8.5). Feel free to reopen if you should come to think of anything more related to this
